src/taipy/core/notification/notifier.py

```diff
@@ -10,34 +10,45 @@
 # specific language governing permissions and limitations under the License.
 
 from queue import Queue
-from typing import List, Tuple
+from typing import Dict, List, Optional
 
-from .event import Event
+from .event import Event, EventEntityType, EventOperation
 from .registration import Registration
 from .topic import Topic
 
 
 class Notifier:
-    _registrations: List[Registration] = []
+    _registrations: Dict[Topic, List[Registration]] = {}  # What if this is a dictionary instead?
 
     @classmethod
-    def register(cls, entity_type, entity_id, operation, attribute_name) -> Tuple[str, Queue]:
-        register_id = "gui"  # TODO generate an id to return so the client can unregister.
-        registration = Registration(register_id, entity_type, entity_id, operation, attribute_name)
-        cls._registrations.append(registration)
-        return register_id, registration.queue
+    def register(
+        cls,
+        entity_type: Optional[str],
+        entity_id: Optional[str],
+        operation: Optional[str],
+        attribute_name: Optional[str],
+    ) -> Registration:
+        registration = Registration(entity_type, entity_id, operation, attribute_name)
+
+        if registrations := cls._registrations.get(registration.topic, None):
+            registrations.append(registration)
+        else:
+            cls._registrations[registration.topic] = [registration]
+
+        return registration
 
     @classmethod
-    def unregister(cls, register_id: str = "gui"):
-        cls._registrations = [reg for reg in cls._registrations if reg.register_id != register_id]
+    def unregister(cls, registration: Registration):
+        if registrations := cls._registrations.get(registration.topic, None):
+            registrations.remove(registration)
+            if len(registrations) == 0:
+                del cls._registrations[registration.topic]
 
     @classmethod
     def publish(cls, event: Event):
-        for registration in cls._registrations:
-            if cls.is_matching(event, registration.topic):
-                registration.queue.put(event)
+        generated_matched_topics_with_event = Topic.generate_topics_from_event(event)
 
-    @classmethod
-    def is_matching(cls, event: Event, topic: Topic) -> bool:
-        # TODO implement logic to see if event matches the topic
-        return True
+        for topic in generated_matched_topics_with_event:
+            if registrations := cls._registrations.get(topic, None):
+                for registration in registrations:
+                    registration.queue.put(event)
```

Comment:
it's certainly easier to maintain and understand. But I think the drawback is that we have to go through all of the topics, this won't be bad if we only have a few topics, but it can be huge if the number of topics is huge (this would be O(N)). While if we generate what topic we should look for from the event, it'd be O(1). It's not necessarily better because we don't have a real use case for this. We might expect the number of topics to be low, so the option you proposed would be better, both performance and readability-wise, but if the number of topics is huge, I think the current implementation would be better. However, I agree that for now, it's better to focus on maintainability rather than trying to optimize it. :D So I'll go with what you proposed

Reply:
To be honest, I did not though about the performance here. You are right that it will be faster when the nb of topics is big. I also agree with focusing first on readability than performance. We first need to measure the performance before optimizing it. Especially because 
when the nb of topics is small, like today, spending time on generating all topics can compensate for the time saved on searching topics. I am not even sure it is faster for now.

Reply:
totally agree with you 😃 let's just focus on readability for now and roll it out, see how it's being used, and if optimization is needed. If we try to optimize it at this point, I think we might over-engineer